tools/mlir_bench/ov_model_gen.py

```diff
@@ -0,0 +1,247 @@
+        return torch.float16
+    if type == 'bf16':
+        return torch.bfloat16
+    assert False, f"Unsupported torch data type {type}"
+
+
+def get_torch_layer(layer: str, sizes: list[int], type: str) -> nn.Module:
+    data_type = get_torch_type(type)
+    if layer == 'linear':
+        assert len(sizes) == 2, "invalid sizes for linear"
+        linear = nn.Linear(*sizes, dtype=data_type)
+        # Generate random weights
+        linear.weight.data.normal_(0, 0.01)
+        linear.bias.data.fill_(0.01)
+        return linear
+    if layer == 'relu':
+        return nn.ReLU()
+    if layer == 'add':
+        return TorchAdd(sizes, data_type)
+    if layer == 'sub':
+        return TorchSub(sizes, data_type)
+    if layer == 'mul':
+        return TorchMul(sizes, data_type)
+    if layer == 'div':
+        return TorchDiv(sizes, data_type)
+    if layer == 'matmul':
+        assert len(sizes) == 3, "invalid sizes for mm"
+        return TorchMatmul(sizes, data_type)
+    assert False, f"Unsupported torch layer type {layer}"
+
+
+def get_layer_name(layer_desc: str) -> str:
+    return layer_desc[0:layer_desc.find('[')]
+
+
+def get_layer_sizes(layer_desc: str) -> list[int]:
+    desc_sizes = layer_desc[layer_desc.find('[')+1:layer_desc.find(']')]
+    return [int(size) for size in filter(None, desc_sizes.split(','))]
+
+
+def parse_layer(layer_desc: str, type: str) -> nn.Module:
+    layer = get_layer_name(layer_desc)
+    sizes = get_layer_sizes(layer_desc)
+    return get_torch_layer(layer, sizes, type)
+
+
+def get_ov_type(type: str) -> ov.Type:
+    if type == 'f32':
+        return ov.Type.f32
+    if type == 'f16':
+        return ov.Type.f16
+    if type == 'bf16':
+        return ov.Type.bf16
+    assert False, f"Unsupported OV data type {type}"
+
+
+def get_layer_inputs(layer_desc: str, is_dynamic: bool):
+    input_sizes = get_layer_sizes(layer_desc)
+    if is_dynamic:
+        input_sizes = [-1] * len(input_sizes)
+
+    layer = get_layer_name(layer_desc)
+
+    if layer == 'matmul':
+        m = input_sizes[0]
+        k = input_sizes[2]
+        return [[m,k]]
+
+    # Needs to be reversed for nn.Linear, does it apply to other layers too?
+    if layer == 'linear':
+        input_sizes.reverse()
+
+    return [input_sizes]
+
+
+def generate_ov_model(layers_desc: str, data_type: str, file_name: str, is_dynamic: bool = False):
+    layers = layers_desc.split()
+    torch_seq = TorchSequential()
+    for layer in layers:
+        module = parse_layer(layer, data_type)
+        torch_seq.append(module)
+
+    input_sizes = get_layer_sizes(layers[0])
+    if len(input_sizes) == 0:
+        print("Invalid input layer sizes")
+        sys.exit(1)
+
+    input_shapes = get_layer_inputs(layers[0], is_dynamic)
+    ov_type = get_ov_type(data_type)
+    inputs = [(ov.PartialShape(shapes), ov_type) for shapes in input_shapes]
+
+    ov_model = ov.convert_model(torch_seq, input=inputs)
+    ov.save_model(ov_model, f"{file_name}")
```

Comment:
Just FYI: `ov.save_model` will partially compress weights from `f32` to `f16`. "Partially" is because for each constant it is decided individually based on the values range in the constant. In the IR it will be represented as two operations: `Constant(f16) -> Convert(f32)`. This is our default behavior for all models in `ov.save_model` to save space on the disk, and it shouldn't affect final inference because this combination of operations will be constant folded during model compilation.

Reply:
It doesn't affect inference precision, it is just a way of weights compression.